README.md

```diff
@@ -34,60 +34,60 @@ Library run by a user with user name "my_user_name" on Linux:
-
-| Function name                 | Example project name           | Value on Linux | Value on Windows | Value on macOS                 |
-| ----------------------------- | ------------------------------ | -------------- | ---------------- | ------------------------------ |
-| `from_unprocessed_string`     | `"FooBar-App"`                 | `"FooBar-App"` | `"FooBar-App"`   | `"FooBar-App"`                 |
-| `from_project_name`           | `"FooBar App"`                 | `"foobar-app"` | `"FooBar App"`   | `"FooBar App"`                 |
-| `from_qualified_project_name` | `"org.foobar-corp.FooBar-App"` | `"foobar-app"` | `"FooBar-App"`   | `"org.foobar-corp.FooBar-App"` |
+| Function name    | Value on Linux                                                                            | Value on Windows                                  | Value on macOS                                        |
+| ---------------- | ----------------------------------------------------------------------------------------- | ------------------------------------------------- | ----------------------------------------------------- |
+| `cache_dir`      | `$XDG_CACHE_HOME/_yourprojectdir_`        or `$HOME/.cache/_yourprojectdir_/`             | `{FOLDERID_LocalAppData}/_yourprojectdir_/cache/` | `$HOME/Library/Caches/_yourprojectdir_/`              |
+| `config_dir`     | `$XDG_CONFIG_HOME/_yourprojectdir_`       or `$HOME/.config/_yourprojectdir_/`            | `{FOLDERID_RoamingAppData}/_yourprojectdir_/`     | `$HOME/Library/Preferences/_yourprojectdir_/`         |
+| `data_dir`       | `$XDG_DATA_HOME/_yourprojectdir_`         or `$HOME/.local/share/_yourprojectdir_/`       | `{FOLDERID_RoamingAppData}/_yourprojectdir_/`     | `$HOME/Library/Application Support/_yourprojectdir_/` |
+| `data_local_dir` | `$XDG_DATA_HOME/_yourprojectdir_`         or `$HOME/.local/share/_yourprojectdir_/`       | `{FOLDERID_LocalAppData}/_yourprojectdir_/`       | `$HOME/Library/Application Support/_yourprojectdir_/` |
+| `runtime_dir`    | `Some($XDG_RUNTIME_DIR/_yourprojectdir_)` or `Some($HOME/.local/share/_yourprojectdir_/)` | `None`                                            | `None`                                                |
+
+The specific value of `_yourprojectdir_` depends on the function used to create the `ProjectDirs` struct.
+Considering the following example values `"org"` (qualifier), `"Baz Corp"` (organization) and `"Foo Bar-App"` (project), the resulting value of _yourprojectdirectory_ will be:
+
+| Function name          | Value on Linux  | Value on Windows         | Value on macOS                 |
+| ---------------------- | --------------- | ------------------------ | ------------------------------ |
+| `from                ` | `"foobar-app"`  | `"Baz Corp/Foo Bar-App"` | `"org.Baz-Corp.Foo-Bar-App"`   |
+| `from_verbatim_values` | `"Foo Bar-App"` | `"Baz Corp/Foo Bar-App"` | `"org.Baz Corp.Foo Bar-App"`   |
```

Comment:
this is a little confusing imho: all used values are used verbatim, but some aren't used at all.

I think part of the confusion would resolve itself if you jotted down (with documentation comments and/or in the README) the purpose of the different factory functions.

imho it looks like the new `from` is an improvement on the previous trio of factory functions, but it should then delegate to a function that just takes a just 1 string that is used verbatim to construct the paths.

Reply:
That's funny, I had a commit lined up that would do exactly that. :-)
Will look into it in the evening again!
Thanks!

Reply:
That's funny, I had a commit lined up that would do exactly that. :-)
Will look into it in the evening again!
Thanks!

Reply:
To be honest, I think the `<company>/<app>` naming scheme is only really used on Windows. What we could do is just have a `from` function which takes a name and on Windows the name is also used for the "company" part of the path. Then there can be a second function which also takes the domain for people who want to explicitly define the domain.

Reply:
True, but the qualifier is also only used on macOS. That's part of the issue: different operating systems require different pieces of information.

Accepting a single name and trying to reverse engineer which part of that name refers to which piece of required information, that's what I did in the beginning, and it was way to complex and unpredictable.

Reply:
Not what I wrote. I said use a single name and just use it for both parts of the qualifier. Because let's face it: if you as an application developer (even developing on MacOS or Windows) don't specify a domain, chances are you don't care and it just picks your app name instead :P